applications/matter_bridge/src/ble_providers/ble_lbs_data_provider.cpp

```diff
@@ -22,20 +26,72 @@ static bt_uuid *sUuidLED = BT_UUID_LBS_LED;
 static bt_uuid *sUuidButton = BT_UUID_LBS_BUTTON;
 static bt_uuid *sUuidCcc = BT_UUID_GATT_CCC;
 
+#ifdef CONFIG_BRIDGE_ONOFF_LIGHT_SWITCH_BRIDGED_DEVICE
+void ProcessCommand(CommandId aCommandId, const EmberBindingTableEntry &aBinding, OperationalDeviceProxy *aDevice, void *aContext)
+{
+	CHIP_ERROR ret = CHIP_NO_ERROR;
+	BindingHandler::BindingData *data = reinterpret_cast<BindingHandler::BindingData *>(aContext);
+
+	auto onSuccess = [dataRef = data](const ConcreteCommandPath &commandPath, const StatusIB &status, const auto &dataResponse) {
+		LOG_DBG("Binding command applied successfully!");
+
+		/* If session was recovered and communication works, reset flag to the initial state. */
+		if (dataRef->CaseSessionRecovered) {
+			dataRef->CaseSessionRecovered = false;
+		}
+	};
+
+	auto onFailure = [dataRef = *data](CHIP_ERROR aError) mutable {
+		BindingHandler::OnInvokeCommandFailure(dataRef, aError);
+	};
+
+	if (aDevice) {
+		/* We are validating connection is ready once here instead of multiple times in each case statement
+		 * below. */
+		VerifyOrDie(aDevice->ConnectionReady());
+	}
+
+	Clusters::OnOff::Commands::Toggle::Type toggleCommand;
+	if (aDevice) {
+		ret = Controller::InvokeCommandRequest(aDevice->GetExchangeManager(),
```

Comment:
I'd think about wrapping this `Controller::` API into the BindingHander. Maybe by doing so, and moving the above to the BindingHander we could simplify the code enough to be able to pass just lambdas as an InvokeCommand argument in the mInvokeCommandCallback calls.